File: frontend/src/api/assistant.js
import api from "./client";


// =================================================
// SEND ASSISTANT MESSAGE
// =================================================

export async function sendAssistantMessage(
  data
) {
  const response = await api.post(
    "/assistant/chat",
    {
      message: data.message,
      conversation_type:
        data.conversation_type || "chat",
    }
  );

  return response.data;
}


// =================================================
// GET ASSISTANT HISTORY
// =================================================

export async function getAssistantHistory(
  limit = 50,
  offset = 0
) {
  const response = await api.get(
    "/assistant/history",
    {
      params: {
        limit,
        offset,
      },
    }
  );

  return response.data;
}


// =================================================
// DELETE ASSISTANT CONVERSATION
// =================================================

export async function deleteAssistantConversation(
  conversationId
) {
  const response = await api.delete(
    `/assistant/history/${conversationId}`
  );

  return response.data;
}

File: backend/app/schemas/assistant.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


# =================================================
# ASSISTANT CHAT REQUEST
# =================================================


class AssistantChatRequest(BaseModel):
    """
    Request sent by the creator to the
    N1MOX30 Personal Creator AI Assistant.
    """

    message: str = Field(
        min_length=1,
        max_length=10000,
    )

    conversation_type: str = Field(
        default="chat",
        max_length=100,
    )


# =================================================
# ASSISTANT CHAT RESPONSE
# =================================================


class AssistantChatResponse(BaseModel):
    """
    Response returned by the N1MOX30
    Personal Creator AI Assistant.
    """

    conversation_id: str

    message: str

    response: str

    provider: str

    conversation_type: str

    action_type: str | None = None

    action_status: str | None = None

    created_at: datetime


# =================================================
# ASSISTANT CONVERSATION ITEM
# =================================================


class AssistantConversationResponse(BaseModel):
    """
    Single conversation record.
    """

    id: str

    role: str

    message: str

    response: str | None = None

    conversation_type: str

    action_type: str | None = None

    action_status: str | None = None

    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


# =================================================
# ASSISTANT CONVERSATION HISTORY
# =================================================


class AssistantConversationHistoryResponse(BaseModel):
    """
    Paginated assistant conversation history.
    """

    total: int

    limit: int

    offset: int

    conversations: list[
        AssistantConversationResponse
    ]
